File: backend/models/Car.js
const mongoose = require('mongoose');

const carSchema = new mongoose.Schema({
  owner: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  name: {
    type: String,
    required: [true, 'Please provide car name'],
    trim: true
  },
  plateNumber: {
    type: String,
    required: [true, 'Please provide car plate number'],
    unique: true,
    trim: true,
    uppercase: true
  },
  type: {
    type: String,
    required: [true, 'Please provide car type'],
    enum: ['Sedan', 'SUV', 'Hatchback', 'Luxury', 'Sports']
  },
  transmission: {
    type: String,
    required: true,
    enum: ['Automatic', 'Manual']
  },
  fuel: {
    type: String,
    required: true,
    enum: ['Petrol', 'Diesel', 'Electric', 'Hybrid']
  },
  seats: {
    type: Number,
    required: true,
    min: 2,
    max: 8
  },
  price: {
    type: Number,
    required: [true, 'Please provide daily rental price'],
    min: 0
  },
  location: {
    type: String,
    required: [true, 'Please provide location']
  },
  available: {
    type: Boolean,
    default: true
  },
  features: [{
    type: String
  }],
  description: {
    type: String,
    required: [true, 'Please provide description']
  },
  images: [{
    type: String
  }],
  rating: {
    type: Number,
    default: 0,
    min: 0,
    max: 5
  },
  approved: {
    type: Boolean,
    default: false
  },
  createdAt: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model('Car', carSchema);

File: backend-java/src/main/java/com/carrental/model/Car.java
package com.carrental.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@Document(collection = "cars")
public class Car {

    @Id
    private String id;
    
    private String ownerId; // Represents the User ObjectId

    private String name;

    @Indexed(unique = true)
    private String plateNumber;

    private String type; // Sedan, SUV, Hatchback, Luxury, Sports
    
    private String transmission; // Automatic, Manual
    
    private String fuel; // Petrol, Diesel, Electric, Hybrid
    
    private Integer seats;
    
    private Double price;
    
    private String location;
    
    private boolean available = true;
    
    private List<String> features;
    
    private String description;
    
    private List<String> images;
    
    private Double rating = 0.0;
    
    private boolean approved = false;
    
    private Date createdAt = new Date();
}
